### PR TITLE
#1051

### PR DESCRIPTION
**Pull request by vestrada**
https://github.com/craftercms/craftercms/issues/1051 - [studio-ui] Allow Studio JS plugins to load from /config/studio relative to the site #1051

### CODE REVIEW

**Review by rart — CHANGES_REQUESTED** [review]
Please apply discussed changes:
- Prefix
- Config lacking properties messages
- When there's a `<plugin/>` property, dismiss the exterior `name` prop and use the internal. 
- Refactor duplicate code on to `Utils.Forms`

**Comment by vestrada** [comment]
Done

File: static-assets/components/cstudio-admin/mods/content-types.js [hunk 1]
```diff
@@ -20,2847 +20,2912 @@ CStudioAdminConsole.contentTypeSelected = "";
...
+                                        YDom.addClass(this.dsourceContainerEl, datasource.getLabel().replace(/\//g, '').replace(/\s+/g, '-').toLowerCase());
+
+
+                                        var dd = new DragAndDropDecorator(this.dsourceContainerEl);
+                                        this.dsourceContainerEl.prototypeDatasource = datasource;
+                                        var iconElt = CStudioAuthoring.Utils.createIcon(datasources[idx], "fa-database");
+                                        this.dsourceContainerEl.insertBefore(iconElt, this.dsourceContainerEl.firstChild);
+                                    }
+                                    catch (e) {
+
```
Comment (rart) [inline static-assets/components/cstudio-admin/mods/content-types.js]: Remove commented out "alert"

File: static-assets/components/cstudio-forms/data-sources/img-desktop-upload.js [hunk 2]
```diff
@@ -53,7 +53,7 @@ YAHOO.extend(CStudioForms.Datasources.ImgDesktopUpload, CStudioForms.CStudioForm
 		
 		var callback = { 
 			success: function(imageData) {
-                var topWin = window.parent.Self;
+                var topWin = window.parent.self;
```
Comment (rart) [inline static-assets/components/cstudio-forms/data-sources/img-desktop-upload.js]: This refers to WCMRootContent or so, please call it it explicitly.

**Review by rart — CHANGES_REQUESTED** [review]
Please address comments

File: static-assets/components/cstudio-admin/mods/content-types.js (hunk 1 above)
Comment (vestrada) [reply static-assets/components/cstudio-admin/mods/content-types.js]: Done

File: static-assets/components/cstudio-forms/data-sources/img-desktop-upload.js (hunk 2 above)
Comment (vestrada) [reply static-assets/components/cstudio-forms/data-sources/img-desktop-upload.js]: Done

**Review by rart — APPROVED** [review]

**Pull request merged**